### PR TITLE
TIP-706: Add PQB "is associated" sorter

### PR DESCRIPTION
**Pull request by damien-carcel**
## Description

This PR adds a sorter on product associations.

## ES Sorter Checklist

- [x] Add sorter integration tests for the field in the PQB
- [ ] ~Add missing integration to the PimCatalogXIntegration tests for sort~ Not needed
- [x] Add the sorter + specs
- [x] Update the reference documentation

## Definition Of Done

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | OK
| Added Behats                      | -
| Added integration tests           | OK
| Changelog updated                 | -
| Review and 2 GTM                  | OK
| Micro Demo to the PO (Story only) | -
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

**Review by SamirBoulil — APPROVED** [review]
GTM Check productIndexing integration tests :trollface:

**Comment by damien-carcel** [comment]
Gna gna gna!

File: src/Pim/Component/Catalog/Normalizer/Indexing/Product/PropertiesNormalizer.php [hunk 1]
```diff
@@ -41,6 +47,9 @@ public function normalize($product, $format = null, array $context = [])
         $data[StandardPropertiesNormalizer::FIELD_CATEGORIES] = $product->getCategoryCodes();
         $data[StandardPropertiesNormalizer::FIELD_GROUPS] = $product->getGroupCodes();
 
+        $data[StandardPropertiesNormalizer::FIELD_IS_ASSOCIATED] = !$product->getAssociations()->isEmpty()
```
Comment (SamirBoulil) [inline src/Pim/Component/Catalog/Normalizer/Indexing/Product/PropertiesNormalizer.php:50]: simplify ^^

File: src/Pim/Component/Catalog/Normalizer/Standard/Product/PropertiesNormalizer.php [hunk 2]
```diff
@@ -28,6 +28,7 @@ class PropertiesNormalizer extends SerializerAwareNormalizer implements Normaliz
     const FIELD_VALUES = 'values';
     const FIELD_CREATED = 'created';
     const FIELD_UPDATED = 'updated';
+    const FIELD_IS_ASSOCIATED = 'is_associated';
```
Comment (SamirBoulil) [inline src/Pim/Component/Catalog/Normalizer/Standard/Product/PropertiesNormalizer.php:31]: Move

**Review by SamirBoulil — APPROVED** [review]
Mob GTM After fix

File: src/Pim/Component/Catalog/Normalizer/Indexing/Product/PropertiesNormalizer.php (hunk 1 above)
Comment (damien-carcel) [reply src/Pim/Component/Catalog/Normalizer/Indexing/Product/PropertiesNormalizer.php:50]: Done in https://github.com/jjanvier/pim-community-dev/pull/126

File: src/Pim/Component/Catalog/Normalizer/Standard/Product/PropertiesNormalizer.php (hunk 2 above)
Comment (damien-carcel) [reply src/Pim/Component/Catalog/Normalizer/Standard/Product/PropertiesNormalizer.php:31]: Done in https://github.com/jjanvier/pim-community-dev/pull/126

**Pull request merged**